# Analiza szczęścia w różnych krajach


Dane zostały pobrane ze strony kaggle : https://www.kaggle.com/datasets/yamaerenay/world-happiness-report-preprocessed?select=2020_report.csv

Dane użyte do raportu to:
- kraje
- wskaźnik szczęścia
- PKB na mieszkańca
- wskaźnik pomocy socjalnej
- wskaźnik zdrowia
- wskaźnik wolności
- wskaźnik hojności
- wskaźnik zaufania do rządu
- kontynet
- pozostałości dystopii czyli najniższy możliwy wskaźnik szczęśćia w danym kraju

#Opis projektu

Szczęście jest pojęciem psychologicznym i filozoficznym, przez każdego interpretowane w indywidualny sposób. Szczęście może wpływać pozytywnie na zdrowie, zmniejszać poziom stresu, zwiększać produktywność czy wpływać pozytywnie na koncentracje i kreatywność.

W tym raporcie zostanie przeanalizowane czy szczęście w danym kraju zależy od czynników takich jak PKB na mieszkańca, zdrowie, zaufanie do rządu, pomoc socjalna, wolność oraz hojność.

Badania do tego raportu zostały przeprowadzone w 138 krajach w roku 2020.

In [0]:
import pyspark.sql.functions as f
from pyspark.sql.types import *

d = spark.read.format('csv').options(header = 'true', inferSchema = 'true', delimiter = ',').load('/FileStore/tables/2020_report.csv')

display(d)

country happiness_score gdp_per_capita social_support health freedom generosity government_trust dystopia_residual continent Finland 7.808700085 1.285189509 1.499525905 0.961271405 0.66231674 0.159670442 0.477857262 2.7628350260000003 Europe Denmark 7.645599842 1.326948524 1.503449202 0.979332566 0.665039897 0.242793396 0.495260328 2.432740688 Europe Switzerland 7.559899807000001 1.39077425 1.4724034069999998 1.040533185 0.62895447 0.269055754 0.407945901 2.35026741 Europe Iceland 7.504499912000001 1.326501608 1.5475674869999998 1.000843406 0.661980748 0.362330228 0.14454077199999998 2.460688114 Europe Norway 7.4879999160000015 1.42420733 1.495172501 1.0080718990000002 0.670200884 0.287985086 0.434100568 2.168266296 Europe Netherlands 7.448900223 1.338946342 1.463645935 0.975675344 0.61362648 0.336317569 0.368569762 2.352117062 Europe Sweden 7.353499889 1.322235227 1.4333477019999998 0.986470461 0.6502977010000001 0.272827893 0.442066371 2.246299267 Europe New Zealand 7.299600123999999 1.242317915 1.48721838 1.008138299 0.646789908 0.325726211 0.461268276 2.128108025 Australia Austria 7.294199944 1.317285538 1.437444925 1.000933528 0.6033688779999999 0.255509764 0.281256139 2.398446083 Europe Luxembourg 7.237500191 1.536676049 1.387528419 0.986442685 0.610137045 0.19595392 0.367041469 2.153700352 Europe Canada 7.23210001 1.301647663 1.4353919030000002 1.022501945 0.644028127 0.28152892 0.35170185600000003 2.195269108 North America Australia 7.222799778 1.310396433 1.477146268 1.022607684 0.621877193 0.324973613 0.335996419 2.129804134 Australia United Kingdom 7.164500237 1.273061037 1.457844973 0.97570008 0.525168657 0.373433441 0.322601646 2.236721992 Europe Israel 7.128600121 1.216463685 1.4032567740000002 1.008052945 0.420699477 0.26686179600000004 0.09989845 2.713358402 Asia Costa Rica 7.121399879 0.981107712 1.374853611 0.939635336 0.6450177429999999 0.131266311 0.096362092 2.953135014 South America Ireland 7.093699932000001 1.446886778 1.470596433 0.975670695 0.587779939 0.295426995 0.373433262 1.943878174 Europe Germany 7.075799942000001 1.314184546 1.368543744 0.972114801 0.564274132 0.252037704 0.309362292 2.295249462 Europe United States 6.9395999910000015 1.37398684 1.404786706 0.831618011 0.534608245 0.298143059 0.152284741 2.344124794 North America Czech Republic 6.9109001160000005 1.212322354 1.40528667 0.89456445 0.505744576 0.046325929 0.049803223 2.796808243 Europe Belgium 6.863500117999999 1.295842767 1.3986775880000002 0.964901149 0.499805421 0.146966159 0.208724052 2.348626614 Europe United Arab Emirates 6.790800095 1.4310864209999998 1.251170993 0.787814438 0.652936101 0.28065598 0.220213518 2.166965961 Asia Malta 6.7727999689999985 1.252513289 1.442956924 0.972042024 0.633239031 0.341180831 0.17886407699999998 1.952012062 Europe France 6.663799762999999 1.268129349 1.458839178 1.029714227 0.5140509010000001 0.112607703 0.227303237 2.053198338 Europe Mexico 6.465000152999999 1.024387479 1.226333499 0.831601024 0.553892553 0.083094485 0.083133668 2.662540197 North America Uruguay 6.440100192999999 1.071000457 1.425081134 0.856928885 0.59426707 0.132143691 0.193425074 2.167276382 South America Saudi Arabia 6.406499863 1.334328532 1.309950113 0.759818137 0.548477471 0.087440684 0.163322315 2.203118801 Asia Spain 6.4008998870000005 1.230535269 1.421099186 1.051343083 0.425983816 0.16530548 0.109579779 1.997088432 Europe Guatemala 6.398900032 0.7538157109999999 1.174267054 0.705952585 0.613146722 0.170611665 0.0983603 2.882723093 South America Italy 6.38740015 1.236396074 1.3472959999999998 1.022504926 0.321305573 0.17026621100000006 0.040145598 2.249505997 Europe Singapore 6.3770999910000015 1.519580126 1.39545691 1.137814283 0.635317206 0.218770906 0.533162236 0.937031746 Asia Brazil 6.375599861 0.952679873 1.363464117 0.766119063 0.483292729 0.131674588 0.106518604 2.571860313 South America Slovenia 6.363399982000001 1.208652496 1.464677691 0.932548046 0.646700144 0.145701498 0.076516323 1.888586521 Europe E

In [0]:
display(d.count())

138

In [0]:
d.groupBy("continent").count().show()

+-------------+-----+
|    continent|count|
+-------------+-----+
|       Europe|   39|
|       Africa|   35|
|North America|    3|
|South America|   20|
|         Asia|   39|
|    Australia|    2|
+-------------+-----+



In [0]:
dh = d
dh.createOrReplaceTempView("hive_database")

In [0]:
%sql
select * from hive_database

country happiness_score gdp_per_capita social_support health freedom generosity government_trust dystopia_residual continent Finland 7.808700085 1.285189509 1.499525905 0.961271405 0.66231674 0.159670442 0.477857262 2.7628350260000003 Europe Denmark 7.645599842 1.326948524 1.503449202 0.979332566 0.665039897 0.242793396 0.495260328 2.432740688 Europe Switzerland 7.559899807000001 1.39077425 1.4724034069999998 1.040533185 0.62895447 0.269055754 0.407945901 2.35026741 Europe Iceland 7.504499912000001 1.326501608 1.5475674869999998 1.000843406 0.661980748 0.362330228 0.14454077199999998 2.460688114 Europe Norway 7.4879999160000015 1.42420733 1.495172501 1.0080718990000002 0.670200884 0.287985086 0.434100568 2.168266296 Europe Netherlands 7.448900223 1.338946342 1.463645935 0.975675344 0.61362648 0.336317569 0.368569762 2.352117062 Europe Sweden 7.353499889 1.322235227 1.4333477019999998 0.986470461 0.6502977010000001 0.272827893 0.442066371 2.246299267 Europe New Zealand 7.299600123999999 1.242317915 1.48721838 1.008138299 0.646789908 0.325726211 0.461268276 2.128108025 Australia Austria 7.294199944 1.317285538 1.437444925 1.000933528 0.6033688779999999 0.255509764 0.281256139 2.398446083 Europe Luxembourg 7.237500191 1.536676049 1.387528419 0.986442685 0.610137045 0.19595392 0.367041469 2.153700352 Europe Canada 7.23210001 1.301647663 1.4353919030000002 1.022501945 0.644028127 0.28152892 0.35170185600000003 2.195269108 North America Australia 7.222799778 1.310396433 1.477146268 1.022607684 0.621877193 0.324973613 0.335996419 2.129804134 Australia United Kingdom 7.164500237 1.273061037 1.457844973 0.97570008 0.525168657 0.373433441 0.322601646 2.236721992 Europe Israel 7.128600121 1.216463685 1.4032567740000002 1.008052945 0.420699477 0.26686179600000004 0.09989845 2.713358402 Asia Costa Rica 7.121399879 0.981107712 1.374853611 0.939635336 0.6450177429999999 0.131266311 0.096362092 2.953135014 South America Ireland 7.093699932000001 1.446886778 1.470596433 0.975670695 0.587779939 0.295426995 0.373433262 1.943878174 Europe Germany 7.075799942000001 1.314184546 1.368543744 0.972114801 0.564274132 0.252037704 0.309362292 2.295249462 Europe United States 6.9395999910000015 1.37398684 1.404786706 0.831618011 0.534608245 0.298143059 0.152284741 2.344124794 North America Czech Republic 6.9109001160000005 1.212322354 1.40528667 0.89456445 0.505744576 0.046325929 0.049803223 2.796808243 Europe Belgium 6.863500117999999 1.295842767 1.3986775880000002 0.964901149 0.499805421 0.146966159 0.208724052 2.348626614 Europe United Arab Emirates 6.790800095 1.4310864209999998 1.251170993 0.787814438 0.652936101 0.28065598 0.220213518 2.166965961 Asia Malta 6.7727999689999985 1.252513289 1.442956924 0.972042024 0.633239031 0.341180831 0.17886407699999998 1.952012062 Europe France 6.663799762999999 1.268129349 1.458839178 1.029714227 0.5140509010000001 0.112607703 0.227303237 2.053198338 Europe Mexico 6.465000152999999 1.024387479 1.226333499 0.831601024 0.553892553 0.083094485 0.083133668 2.662540197 North America Uruguay 6.440100192999999 1.071000457 1.425081134 0.856928885 0.59426707 0.132143691 0.193425074 2.167276382 South America Saudi Arabia 6.406499863 1.334328532 1.309950113 0.759818137 0.548477471 0.087440684 0.163322315 2.203118801 Asia Spain 6.4008998870000005 1.230535269 1.421099186 1.051343083 0.425983816 0.16530548 0.109579779 1.997088432 Europe Guatemala 6.398900032 0.7538157109999999 1.174267054 0.705952585 0.613146722 0.170611665 0.0983603 2.882723093 South America Italy 6.38740015 1.236396074 1.3472959999999998 1.022504926 0.321305573 0.17026621100000006 0.040145598 2.249505997 Europe Singapore 6.3770999910000015 1.519580126 1.39545691 1.137814283 0.635317206 0.218770906 0.533162236 0.937031746 Asia Brazil 6.375599861 0.952679873 1.363464117 0.766119063 0.483292729 0.131674588 0.106518604 2.571860313 South America Slovenia 6.363399982000001 1.208652496 1.464677691 0.932548046 0.646700144 0.145701498 0.076516323 1.888586521 Europe E

In [0]:
%sql
select country, happiness_score
from hive_database
order by happiness_score;

country happiness_score Afghanistan 2.566900015 Zimbabwe 3.2992000580000003 Rwanda 3.312299967 Tanzania 3.476200104 Botswana 3.478899956 Yemen 3.527400017 Malawi 3.538000107 India 3.573299885 Haiti 3.720799923 Zambia 3.759399891 Burundi 3.775300026 Sierra Leone 3.926399946 Egypt 4.151400089 Madagascar 4.165599823 Ethiopia 4.186200142 Togo 4.187200069 Myanmar 4.308000088 Congo (Kinshasa) 4.31099987 Sri Lanka 4.327000141 Mauritania 4.374599934 Tunisia 4.392199993 Chad 4.422699928 Uganda 4.43200016 Palestinian Territories 4.552800179 Liberia 4.557899952 Ukraine 4.56069994 Kenya 4.583000183 Jordan 4.6333999630000005 Iran 4.672399998 Georgia 4.672599792 Armenia 4.676799774 Nigeria 4.724100113 Mali 4.729300021999999 Burkina Faso 4.768700123 Lebanon 4.771500111 Iraq 4.7848000530000006 South Africa 4.814099789 Gabon 4.829299927 Bangladesh 4.832799911 Cambodia 4.8484001160000005 Albania 4.882699966000001 Niger 4.909599781 Guinea 4.949299812 Senegal 4.980800152 Algeria 5.005099773 Venezuela 5.053199768 Cameroon 5.084899902 Morocco 5.094799995 Bulgaria 5.101500034 Turkmenistan 5.119100093999999 China 5.123899937 Turkey 5.131800175 Nepal 5.137199879 Ghana 5.1479997630000005 Azerbaijan 5.164800167 Congo (Brazzaville) 5.194399834 Benin 5.21600008 Ivory Coast 5.233300208999999 Indonesia 5.285600185 Vietnam 5.353499889 Malaysia 5.384300232 Mongolia 5.456200123 Libya 5.488800049 Croatia 5.504700184 Greece 5.514999866 Belarus 5.539899826 Kyrgyzstan 5.541500092000001 Russia 5.546000004 Montenegro 5.54610014 Tajikistan 5.555699825 Moldova 5.607500076 Bosnia and Herzegovina 5.674099922000001 Dominican Republic 5.689199924 Paraguay 5.692100048 Pakistan 5.69329977 Bolivia 5.747499942999999 Serbia 5.77820015 Peru 5.796800137000001 Japan 5.870800018 South Korea 5.872399807000001 Jamaica 5.889800072000001 Portugal 5.9109001160000005 Ecuador 5.925199986 Latvia 5.949999809 Honduras 5.953199862999999 Argentina 5.974699974 Thailand 5.9987998010000005 Hungary 6.000400066 Philippines 6.006000042 Estonia 6.0218000410000005 Kazakhstan 6.0578999520000005 Mauritius 6.101299762999999 Kuwait 6.102099895 Romania 6.123700142000001 Nicaragua 6.13710022 Cyprus 6.15899992 Colombia 6.163400172999999 Poland 6.1862998010000005 Lithuania 6.215499877999999 Bahrain 6.227300167 Chile 6.228499889 Uzbekistan 6.257599831 Slovakia 6.280600071 Panama 6.3048000339999986 Kosovo 6.325200081 El Salvador 6.34829998 Slovenia 6.363399982000001 Brazil 6.375599861 Singapore 6.3770999910000015 Italy 6.38740015 Guatemala 6.398900032 Spain 6.4008998870000005 Saudi Arabia 6.406499863 Uruguay 6.440100192999999 Mexico 6.465000152999999 France 6.663799762999999 Malta 6.7727999689999985 United Arab Emirates 6.790800095 Belgium 6.863500117999999 Czech Republic 6.9109001160000005 United States 6.9395999910000015 Germany 7.075799942000001 Ireland 7.093699932000001 Costa Rica 7.121399879 Israel 7.128600121 United Kingdom 7.164500237 Australia 7.222799778 Canada 7.23210001 Luxembourg 7.237500191 Austria 7.294199944 New Zealand 7.299600123999999 Sweden 7.353499889 Netherlands 7.448900223 Norway 7.4879999160000015 Iceland 7.504499912000001 Switzerland 7.559899807000001 Denmark 7.645599842 Finland 7.808700085

Z tabeli powyżej można przeczytać, że najmniej szczęśliwe kraje to Afganistan, następnie Zimbabwe i Rwanda z wynikami odpowiednio 2.56, 3.30 i 3.31, natomiast najszczęśliwszy kraj to Finlandia, a następnie Dania i Szwajcaria z wynikami odpowiednio 7.81, 7.65 i 7.55. Porównując te dane widać, że różnica pomiędzy najszczęśliwszym krajem a najmniej szczęśliwym wynosi ponad 5 punktów.

In [0]:
%sql
select country, round(happiness_score, 3)
from hive_database
order by happiness_score;

country round(happiness_score, 3) Afghanistan 2.567 Zimbabwe 3.299 Rwanda 3.312 Tanzania 3.476 Botswana 3.479 Yemen 3.527 Malawi 3.538 India 3.573 Haiti 3.721 Zambia 3.759 Burundi 3.775 Sierra Leone 3.926 Egypt 4.151 Madagascar 4.166 Ethiopia 4.186 Togo 4.187 Myanmar 4.308 Congo (Kinshasa) 4.311 Sri Lanka 4.327 Mauritania 4.375 Tunisia 4.392 Chad 4.423 Uganda 4.432 Palestinian Territories 4.553 Liberia 4.558 Ukraine 4.561 Kenya 4.583 Jordan 4.633 Iran 4.672 Georgia 4.673 Armenia 4.677 Nigeria 4.724 Mali 4.729 Burkina Faso 4.769 Lebanon 4.772 Iraq 4.785 South Africa 4.814 Gabon 4.829 Bangladesh 4.833 Cambodia 4.848 Albania 4.883 Niger 4.91 Guinea 4.949 Senegal 4.981 Algeria 5.005 Venezuela 5.053 Cameroon 5.085 Morocco 5.095 Bulgaria 5.102 Turkmenistan 5.119 China 5.124 Turkey 5.132 Nepal 5.137 Ghana 5.148 Azerbaijan 5.165 Congo (Brazzaville) 5.194 Benin 5.216 Ivory Coast 5.233 Indonesia 5.286 Vietnam 5.353 Malaysia 5.384 Mongolia 5.456 Libya 5.489 Croatia 5.505 Greece 5.515 Belarus 5.54 Kyrgyzstan 5.542 Russia 5.546 Montenegro 5.546 Tajikistan 5.556 Moldova 5.608 Bosnia and Herzegovina 5.674 Dominican Republic 5.689 Paraguay 5.692 Pakistan 5.693 Bolivia 5.747 Serbia 5.778 Peru 5.797 Japan 5.871 South Korea 5.872 Jamaica 5.89 Portugal 5.911 Ecuador 5.925 Latvia 5.95 Honduras 5.953 Argentina 5.975 Thailand 5.999 Hungary 6.0 Philippines 6.006 Estonia 6.022 Kazakhstan 6.058 Mauritius 6.101 Kuwait 6.102 Romania 6.124 Nicaragua 6.137 Cyprus 6.159 Colombia 6.163 Poland 6.186 Lithuania 6.215 Bahrain 6.227 Chile 6.228 Uzbekistan 6.258 Slovakia 6.281 Panama 6.305 Kosovo 6.325 El Salvador 6.348 Slovenia 6.363 Brazil 6.376 Singapore 6.377 Italy 6.387 Guatemala 6.399 Spain 6.401 Saudi Arabia 6.406 Uruguay 6.44 Mexico 6.465 France 6.664 Malta 6.773 United Arab Emirates 6.791 Belgium 6.864 Czech Republic 6.911 United States 6.94 Germany 7.076 Ireland 7.094 Costa Rica 7.121 Israel 7.129 United Kingdom 7.165 Australia 7.223 Canada 7.232 Luxembourg 7.238 Austria 7.294 New Zealand 7.3 Sweden 7.353 Netherlands 7.449 Norway 7.488 Iceland 7.504 Switzerland 7.56 Denmark 7.646 Finland 7.809

#Podział na kontynenty

In [0]:
%sql
select continent, round(avg(happiness_score), 3)
from hive_database
group by continent
order by round(avg(happiness_score), 3);

continent round(avg(happiness_score), 3) Africa 4.504 Asia 5.281 South America 5.895 Europe 6.393 North America 6.879 Australia 7.261

Databricks visualization. Run in Databricks to view.

Z tego wykresu można odczytać, że najszczęśliwszy kontynent to Australia, następnie Ameryka Północna oraz Europa. Pozostałe trzy kontynenty średnio przyjęły wartości poniżej 6. Może być to związane z podziałem gospodarczym na bogatą Północ i biedne Południe.

In [0]:
%sql
select continent, round(avg(happiness_score), 3), round(avg(gdp_per_capita), 3)
from hive_database
group by continent
order by round(avg(happiness_score), 3);

continent round(avg(happiness_score), 3) round(avg(gdp_per_capita), 3) Africa 4.504 0.51 Asia 5.281 0.908 South America 5.895 0.824 Europe 6.393 1.178 North America 6.879 1.233 Australia 7.261 1.276

Databricks visualization. Run in Databricks to view.

Ten wykres przedstawia zależność pomiędzy średnim PKB na mieszkańca na danych kontynentach, co potwierdza tezę powyżej, że szczęście może zależeć od PKB na mieszkańca oraz że kraje bogatej Północy są szczęśliwsze.

In [0]:
%sql
select continent, round(avg(happiness_score), 3), round(avg(freedom), 3), round(avg(health), 3), round(avg(social_support), 3), round(avg(generosity), 3)
from hive_database
group by continent
order by round(avg(happiness_score), 3);

continent round(avg(happiness_score), 3) round(avg(freedom), 3) round(avg(health), 3) round(avg(social_support), 3) round(avg(generosity), 3) Africa 4.504 0.391 0.428 0.885 0.182 Asia 5.281 0.479 0.725 1.159 0.208 South America 5.895 0.51 0.756 1.239 0.157 Europe 6.393 0.492 0.896 1.373 0.176 North America 6.879 0.578 0.895 1.356 0.221 Australia 7.261 0.634 1.015 1.482 0.325

Z tej tabeli można wywnioskować, że kontynenty z wyższym średnim wskaźnikiem szczęścia, również mają wyższe średnie wskaźniki wolności, zdrowia, hojności oraz pomocy społecznej. Potwierdza to tezę, że szczęścia zależy od tych czynników.

In [0]:
%sql
SELECT continent, min(dystopia_residual), max(dystopia_residual), round(avg(happiness_score), 3)
FROM hive_database
group by continent;

continent min(dystopia_residual) max(dystopia_residual) round(avg(happiness_score), 3) Europe 1.252669334 2.796808243 6.393 Africa 0.257240534 3.440809727 4.504 North America 2.195269108 2.662540197 6.879 South America 1.619916916 2.953135014 5.895 Asia 0.611288548 2.9768767360000004 5.281 Australia 2.128108025 2.129804134 7.261

Tabela ta obrazuje wskaźnik pozostałości dystopii, czyli najniższy możliwy wskaźnik szczęścia na danym kontynencie.

In [0]:
d_rounded = d.select(f.round("happiness_score", 3).alias("happiness_score"), "country", "continent", f.round("gdp_per_capita", 3).alias("gdp_per_capita"), f.round("social_support", 3).alias("social_support"), f.round("health", 3).alias("health"), f.round("freedom", 3).alias("freedom"), f.round("government_trust", 3).alias("government_trust"), f.round("dystopia_residual", 3).alias("dystopia_residual"), f.round("generosity", 3).alias("generosity"))

In [0]:
d_rounded.select("country", d_rounded["happiness_score"]).show()

+--------------+---------------+
|       country|happiness_score|
+--------------+---------------+
|       Finland|          7.809|
|       Denmark|          7.646|
|   Switzerland|           7.56|
|       Iceland|          7.504|
|        Norway|          7.488|
|   Netherlands|          7.449|
|        Sweden|          7.353|
|   New Zealand|            7.3|
|       Austria|          7.294|
|    Luxembourg|          7.238|
|        Canada|          7.232|
|     Australia|          7.223|
|United Kingdom|          7.165|
|        Israel|          7.129|
|    Costa Rica|          7.121|
|       Ireland|          7.094|
|       Germany|          7.076|
| United States|           6.94|
|Czech Republic|          6.911|
|       Belgium|          6.864|
+--------------+---------------+
only showing top 20 rows



# Podział na trzy grupy krajów ze względu na poziom szczęścia

In [0]:

d1 = d.withColumn("Poziom szczęscia w krajach", 
                 f.when((d.happiness_score >= 2) & (d.happiness_score < 5), "Najmniej szczęśliwe kraje")\
                 .when((d.happiness_score >= 5) & (d.happiness_score < 7), "Średnio szczęśliwe kraje")\
                 .when((d.happiness_score >= 7), "Najbardziej szczęśliwe kraje"))
d3 = d1.groupBy("Poziom szczęscia w krajach").count()
d3.show(3, False)
display(d3)

+----------------------------+-----+
|Poziom szczęscia w krajach  |count|
+----------------------------+-----+
|Najbardziej szczęśliwe kraje|17   |
|Średnio szczęśliwe kraje    |77   |
|Najmniej szczęśliwe kraje   |44   |
+----------------------------+-----+



Poziom szczęscia w krajach count Najbardziej szczęśliwe kraje 17 Średnio szczęśliwe kraje 77 Najmniej szczęśliwe kraje 44

Databricks visualization. Run in Databricks to view.

Zbiór 138 krajów został podzielony na trzy kategorie ze względu na poziom szczęścia: do 5, od 5 do 7 i od 7. Jak widać najwięcej krajów, to kraje średnio szczęśliwe. Grupa ta wynosi 77 krajów.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("social_support"), 3).alias("social_support"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)

+----------------------------+--------------+-------+
|Poziom szczęscia w krajach  |social_support|country|
+----------------------------+--------------+-------+
|Najbardziej szczęśliwe kraje|1.454         |17     |
|Średnio szczęśliwe kraje    |1.258         |77     |
|Najmniej szczęśliwe kraje   |0.908         |44     |
+----------------------------+--------------+-------+



Poziom szczęscia w krajach social_support country Najbardziej szczęśliwe kraje 1.454 17 Średnio szczęśliwe kraje 1.258 77 Najmniej szczęśliwe kraje 0.908 44

Databricks visualization. Run in Databricks to view.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("gdp_per_capita"), 3).alias("gdp_per_capita"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)

+----------------------------+--------------+-------+
|Poziom szczęscia w krajach  |gdp_per_capita|country|
+----------------------------+--------------+-------+
|Najbardziej szczęśliwe kraje|1.315         |17     |
|Średnio szczęśliwe kraje    |0.984         |77     |
|Najmniej szczęśliwe kraje   |0.541         |44     |
+----------------------------+--------------+-------+



Poziom szczęscia w krajach gdp_per_capita country Najbardziej szczęśliwe kraje 1.315 17 Średnio szczęśliwe kraje 0.984 77 Najmniej szczęśliwe kraje 0.541 44

Databricks visualization. Run in Databricks to view.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("health"), 3).alias("health"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)

+----------------------------+------+-------+
|Poziom szczęscia w krajach  |health|country|
+----------------------------+------+-------+
|Najbardziej szczęśliwe kraje|0.992 |17     |
|Średnio szczęśliwe kraje    |0.776 |77     |
|Najmniej szczęśliwe kraje   |0.488 |44     |
+----------------------------+------+-------+



Poziom szczęscia w krajach health country Najbardziej szczęśliwe kraje 0.992 17 Średnio szczęśliwe kraje 0.776 77 Najmniej szczęśliwe kraje 0.488 44

Databricks visualization. Run in Databricks to view.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("freedom"), 3).alias("freedom"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)

+----------------------------+-------+-------+
|Poziom szczęscia w krajach  |freedom|country|
+----------------------------+-------+-------+
|Najbardziej szczęśliwe kraje|0.613  |17     |
|Średnio szczęśliwe kraje    |0.486  |77     |
|Najmniej szczęśliwe kraje   |0.384  |44     |
+----------------------------+-------+-------+



Poziom szczęscia w krajach freedom country Najbardziej szczęśliwe kraje 0.613 17 Średnio szczęśliwe kraje 0.486 77 Najmniej szczęśliwe kraje 0.384 44

Databricks visualization. Run in Databricks to view.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("generosity"), 3).alias("generosity"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)

+----------------------------+----------+-------+
|Poziom szczęscia w krajach  |generosity|country|
+----------------------------+----------+-------+
|Najbardziej szczęśliwe kraje|0.273     |17     |
|Średnio szczęśliwe kraje    |0.163     |77     |
|Najmniej szczęśliwe kraje   |0.196     |44     |
+----------------------------+----------+-------+



Poziom szczęscia w krajach generosity country Najbardziej szczęśliwe kraje 0.273 17 Średnio szczęśliwe kraje 0.163 77 Najmniej szczęśliwe kraje 0.196 44

Databricks visualization. Run in Databricks to view.

In [0]:
d3 = d1.groupBy("Poziom szczęscia w krajach").agg(f.round(f.avg("government_trust"), 3).alias("government_trust"), f.count("*").alias("country"))
d3.show(3, False)
display(d3)


+----------------------------+----------------+-------+
|Poziom szczęscia w krajach  |government_trust|country|
+----------------------------+----------------+-------+
|Najbardziej szczęśliwe kraje|0.339           |17     |
|Średnio szczęśliwe kraje    |0.102           |77     |
|Najmniej szczęśliwe kraje   |0.104           |44     |
+----------------------------+----------------+-------+



Poziom szczęscia w krajach government_trust country Najbardziej szczęśliwe kraje 0.339 17 Średnio szczęśliwe kraje 0.102 77 Najmniej szczęśliwe kraje 0.104 44

Databricks visualization. Run in Databricks to view.

Na wszystkich powyższych wykresach pokazujących podział krajów na trzy grupy ze względu na poziom szczęścia. Widać, że kraje z najwyższym poziomem szczęścia, mają najwyższe wszystkie wskaźniki. Natomiast kraje z najniższym poziomem trzecim mają najniższe wskaźniki dla wsparcia socjalnego, PKB na mieszkańca, zdrowie, wolność. Jednak dla wskaźników zaufania do rządu oraz hojności przyjmują one wyższe wartości niż dla krajów ze średnim poziomem szczęścia, chociaż w odniesieniu do wskaźnika zaufania do rządu różnica jest niewielka.

# Podział 10 najszczęśliwszych krajów i 10 najmniej szczęśliwych krajów

In [0]:
min_score = d_rounded.agg(f.min("happiness_score")).collect()[0][0]
max_score = d_rounded.agg(f.max("happiness_score")).collect()[0][0]

min_country = d_rounded.filter(d_rounded.happiness_score == min_score).select("country").collect()[0][0]
max_country = d_rounded.filter(d_rounded.happiness_score == max_score).select("country").collect()[0][0]


results = spark.createDataFrame([("Minimum", min_score, min_country),
                                  ("Maximum", max_score, max_country)],
                                 ["Type", "Score", "Country"])
display(results)

Type Score Country Minimum 2.567 Afghanistan Maximum 7.809 Finland

In [0]:
d_sorted1 = d_rounded.sort(d_rounded.happiness_score.asc())
d4 = d_sorted1.limit(10)
display(d4.select("happiness_score", "country", "continent"))

d_sorted2 = d_rounded.sort(d_rounded.happiness_score.desc())
d5 = d_sorted2.limit(10)
display(d5.select("happiness_score", "country", "continent"))



happiness_score country continent 2.567 Afghanistan Asia 3.299 Zimbabwe Africa 3.312 Rwanda Africa 3.476 Tanzania Africa 3.479 Botswana Africa 3.527 Yemen Asia 3.538 Malawi Africa 3.573 India Asia 3.721 Haiti South America 3.759 Zambia Africa

Databricks visualization. Run in Databricks to view.

happiness_score country continent 7.809 Finland Europe 7.646 Denmark Europe 7.56 Switzerland Europe 7.504 Iceland Europe 7.488 Norway Europe 7.449 Netherlands Europe 7.353 Sweden Europe 7.3 New Zealand Australia 7.294 Austria Europe 7.238 Luxembourg Europe

Databricks visualization. Run in Databricks to view.

Z dwóch powyższych wykresów można odczytać, że 10 najmniej szczęśliwych krajów to w większości kraje Afryki, trzy z nich są z Azji oraz jeden z Ameryki Południowej. Natomiast 9 krajów z 10 najszczęśliwszych to kraje europejskie oraz jeden z Australii, czyli Nowa Zelandia. To również potwierdza, że kraje bogatej Północy mają wyższy wskaźniki poziomu szczęścia niż biednego Południa.

In [0]:
display(d4.select("happiness_score", "country", "gdp_per_capita", "government_trust"))
display(d5.select("happiness_score", "country", "gdp_per_capita", "government_trust"))

happiness_score country gdp_per_capita government_trust 2.567 Afghanistan 0.301 0.001 3.299 Zimbabwe 0.426 0.081 3.312 Rwanda 0.343 0.486 3.476 Tanzania 0.457 0.204 3.479 Botswana 0.998 0.102 3.527 Yemen 0.393 0.087 3.538 Malawi 0.177 0.132 3.573 India 0.731 0.106 3.721 Haiti 0.285 0.162 3.759 Zambia 0.537 0.087

happiness_score country gdp_per_capita government_trust 7.809 Finland 1.285 0.478 7.646 Denmark 1.327 0.495 7.56 Switzerland 1.391 0.408 7.504 Iceland 1.327 0.145 7.488 Norway 1.424 0.434 7.449 Netherlands 1.339 0.369 7.353 Sweden 1.322 0.442 7.3 New Zealand 1.242 0.461 7.294 Austria 1.317 0.281 7.238 Luxembourg 1.537 0.367

In [0]:
display(d4.select("happiness_score", "country", "social_support", "freedom", "health", "generosity"))
display(d5.select("happiness_score", "country", "social_support", "freedom", "health", "generosity"))

happiness_score country social_support freedom health generosity 2.567 Afghanistan 0.356 0.0 0.266 0.135 3.299 Zimbabwe 1.048 0.377 0.375 0.151 3.312 Rwanda 0.523 0.604 0.572 0.236 3.476 Tanzania 0.873 0.509 0.443 0.272 3.479 Botswana 1.086 0.509 0.494 0.033 3.527 Yemen 1.177 0.244 0.415 0.095 3.538 Malawi 0.53 0.487 0.446 0.213 3.573 India 0.644 0.581 0.541 0.237 3.721 Haiti 0.647 0.169 0.374 0.464 3.759 Zambia 0.896 0.491 0.364 0.251

happiness_score country social_support freedom health generosity 7.809 Finland 1.5 0.662 0.961 0.16 7.646 Denmark 1.503 0.665 0.979 0.243 7.56 Switzerland 1.472 0.629 1.041 0.269 7.504 Iceland 1.548 0.662 1.001 0.362 7.488 Norway 1.495 0.67 1.008 0.288 7.449 Netherlands 1.464 0.614 0.976 0.336 7.353 Sweden 1.433 0.65 0.986 0.273 7.3 New Zealand 1.487 0.647 1.008 0.326 7.294 Austria 1.437 0.603 1.001 0.256 7.238 Luxembourg 1.388 0.61 0.986 0.196

Ostatnie cztery tabele przedstawiają, poziomy różnych wskaźników wpływających na wskaźnik szczęścia w 10 krajach z najwyższym wskaźnikiem szczęścia oraz w 10 krajach z najniższym wskaźnikiem szczęścia. Największa różnica jest widoczna przy wskaźnikach PKB na mieszkańca oraz zdrowia, które są dużo wyższy dla krajów najszczęśliwszych. Również dla krajów najszczęśliwszych wskaźnik pomocy socjalnej jest wyższy. Wskaźniki wolności i hojności są na podobnym poziomie dla obu grup, jednak pomimo tego wyższe wskaźniki przyjmuje w większości w krajach najszczęśliwszych. Jeśli weźmie się pod uwagę wskaźnik zaufania do rządu, to oprócz Islandii gdzie ten wskaźnik jest niższy, to w pozostałych krajach z najwyższym poziomem szczęścia jest on wyższy w porównaniu do krajów z niższym poziomem szczęścia (oprócz Rwandy gdzie jest on na wysokim poziomie).

# Podsumowanie
Biorąc pod uwagę, powyższe dane można zauważyć zależność pomiędzy poziomem szczęścia a resztą wskaźników wykorzystanych w trakcie analizy. Oznacza to, że im wyższy poziom wskaźników PKB na mieszkańca, zdrowia (gdzie zależność przy tych dwóch wskaźnikach była najbardziej widoczna), wolności, hojności, zaufania do rządu, pomocy socjalnej, tym wyższy poziom szczęścia.

Najlepiej tę zależność można zauważyć przy podziale krajów ze względu na kontynent. Gdzie wskaźniki wykazują liniową zależność względem wskaźnika szczęścia. Gdy wskaźniki rosną, to również poziom szczęścia rośnie. Również te dane wskazują, że kraje bogatej Północy, czyli kraje lepiej rozwinięte, zasobne, niemające problemów skrajnego ubóstwa czy głodu, są krajami szczęśliwszymi niż kraje biednego Południa, gdzie te problemy występują. Co oczywiście jest zrozumiałe i było do przewidzenia.

W podziale na trzy grupy ze względu na poziom szczęścia. Grupa średnio szczęśliwych jest najliczniejsza. Krajów najszczęśliwszych jest najmniej, jednak to właśnie dla nich wszystkie wskaźniki porównujące były najwyższe. Jedynie dla najmniej szczęśliwych krajów dwa wskaźniki były na średnim poziomie, chociaż z niewielką różnicą.

Ostatni podział obrazuje zależność między wskaźnikami a poziomem szczęścia indywidualnie dla każdego kraju (10 najszczęśliwszych krajów i 10 najmniej szczęśliwych). Tutaj również można wywnioskować, że im wyższe są wskaźniki, tym poziom szczęścia jest wyższy, choć zależność ta dla niektórych wskaźników już w tym wypadku nie była, aż tak widoczna.

Podsumowując, szczęście w danym kraju zależy od czynników takich jak PKB na mieszkańca, zdrowie, zaufanie do rządu, pomoc socjalna, wolność oraz hojność. Również położenie geograficzne ma znaczenie ze względu na dostęp do zasobów oraz rozwoju gospodarczego.